packages/sdk/src/signingcosmwasmclient.ts

```diff
@@ -80,14 +95,15 @@ export class SigningCosmWasmClient extends CosmWasmClient {
     return super.getAccount(address || this.senderAddress);
   }
 
-  /** Uploads code and returns a code ID */
-  public async upload(wasmCode: Uint8Array, memo = ""): Promise<number> {
+  /** Uploads code and returns a receipt, including the code ID */
+  public async upload(wasmCode: Uint8Array, memo = ""): Promise<UploadReceipt> {
+    const compressed = pako.gzip(wasmCode, { level: 9 });
```

Comment:
This is good. We can merge like this, but I would check the magic byte first, so this works for both wasm and gzip wasm.

I will link the go code. But that can be a follow up pr

Reply:
> so this works for both wasm and gzip wasm.

Why would the user ever want to create the gzip version manually? There are already multiple different representations of the contract that the user needs to handle (source, build, optimized built), and I suggest to leave the compression up to clients like CosmWasmClient and wasm-cli. As a consequence `wasmCode` is always the original wasm.